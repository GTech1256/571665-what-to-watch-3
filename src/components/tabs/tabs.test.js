import React from "react";
import renderer from "react-test-renderer";
import Tabs from "./tabs.jsx";
import Tab from "../tab/tab.jsx";


const mock = {
  titles: [`1`, `2`, `3`],
  childrens: [
    <p key="1">1</p>,
    <p key="2">2</p>,
    <p key="3">3</p>,
  ]
};

it(`<Tabs /> shoult render correct`, () => {
  const tree = renderer.create(
      <Tabs>
        {mock.titles.map((it, idx) => (
          <Tab key={it} title={it}>
            {mock.childrens[idx]}
          </Tab>
        ))}
      </Tabs>
  );

  expect(tree).toMatchSnapshot();
});
