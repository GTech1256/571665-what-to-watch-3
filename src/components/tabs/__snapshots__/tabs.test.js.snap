// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> shoult render correct 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
        >
          1
        </a>
      </li>
      <li
        className="movie-nav__item "
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
        >
          2
        </a>
      </li>
      <li
        className="movie-nav__item "
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
        >
          3
        </a>
      </li>
    </ul>
  </nav>
  <p>
    1
  </p>
</div>
`;
